libcudacxx/test/libcudacxx/std/numerics/bit/bit.cast/bit_cast.pass.cpp

```diff
@@ -0,0 +1,348 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+// <cuda/std/bit>
+//
+// template<class To, class From>
+//   constexpr To bit_cast(const From& from) noexcept;
+
+#include <cuda/std/array>
+#include <cuda/std/bit>
+#include <cuda/std/cassert>
+#include <cuda/std/cmath>
+#include <cuda/std/cstdint>
+#include <cuda/std/limits>
+
+#include "test_macros.h"
+
+__host__ __device__ cuda::std::size_t memcmp(void* lhs, void* rhs, size_t bytes) noexcept
+{
+  const unsigned char* clhs = (const unsigned char*) lhs;
+  const unsigned char* crhs = (const unsigned char*) rhs;
+
+  for (; bytes > 0; --bytes)
+  {
+    if (*clhs++ != *crhs++)
+    {
+      return clhs[-1] < crhs[-1] ? -1 : 1;
+    }
+  }
+  return 0;
+}
+
+// cuda::std::bit_cast does not preserve padding bits, so if T has padding bits,
+// the results might not memcmp cleanly.
+template <bool HasUniqueObjectRepresentations = true, typename T>
+__host__ __device__ void test_roundtrip_through_buffer(T from)
+{
+  struct Buffer
+  {
+    char buffer[sizeof(T)];
+  };
+  Buffer middle  = cuda::std::bit_cast<Buffer>(from);
+  T to           = cuda::std::bit_cast<T>(middle);
+  Buffer middle2 = cuda::std::bit_cast<Buffer>(to);
+
+  assert((from == to) == (from == from)); // because NaN
+
+  _CCCL_IF_CONSTEXPR (HasUniqueObjectRepresentations)
+  {
+    assert(memcmp(&from, &middle, sizeof(T)) == 0);
+    assert(memcmp(&to, &middle, sizeof(T)) == 0);
+    assert(memcmp(&middle, &middle2, sizeof(T)) == 0);
+  }
+}
+
+template <bool HasUniqueObjectRepresentations = true, typename T>
+__host__ __device__ void test_roundtrip_through_nested_T(T from)
+{
+  struct Nested
+  {
+    T x;
+  };
+  static_assert(sizeof(Nested) == sizeof(T));
+
+  Nested middle  = cuda::std::bit_cast<Nested>(from);
+  T to           = cuda::std::bit_cast<T>(middle);
+  Nested middle2 = cuda::std::bit_cast<Nested>(to);
+
+  assert((from == to) == (from == from)); // because NaN
+
+  _CCCL_IF_CONSTEXPR (HasUniqueObjectRepresentations)
+  {
+    assert(memcmp(&from, &middle, sizeof(T)) == 0);
+    assert(memcmp(&to, &middle, sizeof(T)) == 0);
+    assert(memcmp(&middle, &middle2, sizeof(T)) == 0);
+  }
+}
+
+template <typename Intermediate, bool HasUniqueObjectRepresentations = true, typename T>
+
+__host__ __device__ void test_roundtrip_through(T from)
```

Comment:
Suggestion: delete empty line:
```suggestion
template <typename Intermediate, bool HasUniqueObjectRepresentations = true, typename T>
__host__ __device__ void test_roundtrip_through(T from)
```